ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/s3_utilities.py

```diff
@@ -0,0 +1,358 @@
+# output folder and for those files with a .csv file extension, and pushing
+# them to the appropriate folder specified by a parameter passed to the function.
+# The logic also ignores the export_summary.csv file which is generated by syntha
+# but not used by the BFD ETL pipeline.
+#
+# Param: synthea_output_dir : unix filesystem directory that the synthea generation
+# process  wrote its output files.
+# Param: s3_folder : BFD S3 Bucket/folder that the .CSV files will be uploaded to.
+#
+# Raises a python exception if failure to upload file.
+def upload_rif_files(synthea_output_dir, s3_bucket, s3_folder):
+    for fn in os.listdir(synthea_output_dir):
+        ## ignore the export_summary.csv
+        if fn.startswith("export_summary"):
+            continue
+        tmp_ext = fn.split('.')[1:]
+        if len(tmp_ext) > 0 and tmp_ext[0] == 'csv':
+            try:
+                local_fn = synthea_output_dir + fn
+                remote_fn = s3_folder + "/" + fn
+                print(f"{local_fn} ==> S3: {s3_bucket} : {remote_fn}")
+                s3_client.upload_file(local_fn, s3_bucket, remote_fn)
+            except botocore.exceptions.ClientError as e:
+                if e.response['Error']['Code'] == "404":
+                    print("The object does not exist.")
+                else:
+                    raise
+
+# Function to upload synthea generated manifest.xml file to the BFD ETL /Incoming
+# S3 bucket folder. This occurs after successfully uploading beneficiary and
+# claims .CSV files to the same S3 bucket; the reason for this is that when the
+# BFD ETL process detects the manifest.xml file, it triggers processing of the
+# .CSV file(s).
+#
+# Param: synthea_output_dir : unix filesystem directory that the synthea generation
+# process  wrote its output files.
+# Param: s3_folder : BFD S3 Bucket/folder that the .CSV files will be uploaded to.
+#
+# Raises a python exception if failure to upload file.
+def upload_manifest_file(synthea_output_dir, s3_bucket, s3_folder):
+    local_fn = synthea_output_dir + "manifest.xml"
+    remote_fn = s3_folder + "/0_manifest.xml"
+    
+    if os.path.exists(local_fn):
+        print(f"S3 upload: {local_fn}, bucket: {s3_bucket}, remote_fn: {remote_fn}")
+        try:
+            s3_client.upload_file(local_fn, s3_bucket, remote_fn)
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == "404":
+                print("The object does not exist.")
+            else:
+                raise
+    else:
+        raise f"Failed to find file: {local_fn}"
+
+# Function to upload synthea generated manifest.xml file to the BFD ETL /Incoming
+# S3 bucket folder. This occurs after successfully uploading beneficiary and
+# claims .CSV files to the same S3 bucket; the reason for this is that when the
+# BFD ETL process detects the manifest.xml file, it triggers processing of the
+# .CSV file(s).
+#
```

Comment:
Looks like a copy/paste mistake from the previous function, should describe this function

Reply:
fixed.

Reply:
Also, notice that I made some changes to the validation script; turns out that based on the randomness that Synthea performs, it is actually possible for some claim types to not be generated. So I protect vs. that in the validation script.

Reply:
Yup, that is true. Generally you'll only see that on extremely small trial runs like you're doing, but might as well cover that case.